### PR TITLE
[PC-8075] block id check maintenance

### PR DESCRIPTION
**Pull request by marionnousvalentin**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-8075

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |          Screenshot_20210510-174248             |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/auth/signup/VerifyEligibility.tsx [hunk 1]
```diff
@@ -17,14 +18,16 @@ type Props = StackScreenProps<RootStackParamList, 'VerifyEligibility'>
 export function VerifyEligibility(props: Props) {
   const { navigate } = useNavigation<UseNavigationType>()
   const deposit = useDepositAmount()
+  const { email, licenceToken } = props.route.params
+
+  const navigateToCheckId = useNavigateToCheckId({ email, licenceToken })
```
[the remark this answers is not in the document]
Comment (marionnousvalentin) [reply src/features/auth/signup/VerifyEligibility.tsx]: Yes, bien sur ! J'ai changé le wording partout !

File: src/features/auth/signup/VerifyEligibility.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import styled from 'styled-components/native'
 
 import { useDepositAmount } from 'features/auth/api'
 import { RootStackParamList, UseNavigationType } from 'features/navigation/RootNavigator'
+import { useNavigateToCheckId } from 'features/navigation/useNavigateToCheckId'
```
[the remark this answers is not in the document]
Comment (marionnousvalentin) [reply src/features/auth/signup/VerifyEligibility.tsx]: j'ai mis la modale et le hook dans une sous feature de auth/signup

File: src/features/navigation/useNavigateToCheckId.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  licenceToken,
+}: {
+  email: string
+  licenceToken: string
+}) => {
+  const { navigate } = useNavigation<UseNavigationType>()
+
+  return () => {
+    navigate('IdCheck', { email, licenceToken })
+  }
```
[the remark this answers is not in the document]
Comment (marionnousvalentin) [reply src/features/navigation/useNavigateToCheckId.ts]: bonne idée, c'est plus adapté, j'ai changé ça

File: src/features/auth/signup/BlockIdCheck.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { Close } from 'ui/svg/icons/Close'
+import { HappyFace } from 'ui/svg/icons/HappyFace'
+import { ColorsEnum, Spacer, getSpacing, Typo } from 'ui/theme'
+
+interface BlockIdCheckModalProps {
+  visible: boolean
+  dismissModal: () => void
+}
+
+export const BlockIdCheckModal = (props: BlockIdCheckModalProps) => {
```
[the remark this answers is not in the document]
Comment (marionnousvalentin) [reply src/features/auth/signup/BlockIdCheck.tsx]: yes, j'ai renommé la modale, c'est plus clair

File: src/features/navigation/useNavigateToCheckId.ts [hunk 5]
```diff
@@ -1,17 +1,25 @@
 import { useNavigation } from '@react-navigation/native'
 
+import { useAppSettings } from 'features/auth/settings'
 import { UseNavigationType } from 'features/navigation/RootNavigator'
 
 export const useNavigateToCheckId = ({
   email,
   licenceToken,
+  showIdCheckBlockingModal,
```
[the remark this answers is not in the document]
Comment (marionnousvalentin) [reply src/features/navigation/useNavigateToCheckId.ts]: j'ai changé le wording de l'input du hook, c'est plus propre comme ça

**Review by blixit — APPROVED** [review]

**Pull request merged**